src/Xamarin.Android.Tools.AnnotationSupport/AndroidAnnotationsSupport.cs

```diff
@@ -24,7 +27,79 @@ public static IList<AnnotatedItem> ParseArchive (string file)
 		static IEnumerable<AnnotatedItem> ParseArchiveEntry (ZipArchiveEntry entry)
 		{
 			using (var s = entry.Open ())
-				return XDocument.Load (s).Root.Elements ("item").Select (e => new AnnotatedItem (e));
+				return SafeXmlLoad (s, entry.FullName).Root.Elements ("item").Select (e => new AnnotatedItem (e));
+		}
+
+		static XDocument SafeXmlLoad (Stream s, string fileName)
+		{
+			// We must save to a temporary stream because the stream doesn't support seeking and should the
+			// parsing fail we won't be able to go back to its beginnig to reparse it.
+			var ms = new MemoryStream ();
+			s.CopyTo (ms);
+			ms.Seek (0, SeekOrigin.Begin);
+
+			try {
+				return XDocument.Load (ms);
+			} catch (XmlException ex) {
+				Console.Error.WriteLine ($"Warning: failed to load annotation document '{fileName}' directly from the annotations archive. {ex.Message}");
+				Console.Error.WriteLine ("Attempting to fix up and reload");
+			}
+
+			Stream ns = null;
+			try {
+				ns = FixAnnotationXML (ms);
+			} catch (Exception ex) {
+				SafeDispose (ms);
+				SafeDispose (ns);
+				throw new InvalidOperationException ($"Failed to fix up invalid XML in annotation document '{fileName}'. {ex.Message}", ex);
+			}
+
+			try {
+				return XDocument.Load (ns);
+			} finally {
+				SafeDispose (ms);
+				SafeDispose (ns);
+			}
+		}
+
+		static void SafeDispose (Stream ms)
+		{
+			try {
+				ms?.Dispose ();
+			} catch {
+				// ignore
+			}
+		}
```

Comment:
I would think `using` blocks, you could avoid the need for this method entirely. The empty try-catch is the part we shouldn't really need.

Was there some trouble with a `Stream` getting double-disposed? It might be there is just a place you need to use an overload of `leaveOpen: true`.

Reply:
The streams in question aren't placed in using blocks because I don't want to nest the blocks too much and the streams *may* be null. The method provides for cleaner code and the empty `catch` is to discard any and all errors in a place where we really don't care about any errors.

Reply:
I don't see how the streams would be `null` if `using` was used?

It seems cleaner to me to just do:
```csharp
	static XDocument SafeXmlLoad (Stream s, string fileName)
	{
		// We must save to a temporary stream because the stream doesn't support seeking and should the
		// parsing fail we won't be able to go back to its beginnig to reparse it.
		using (var ms = new MemoryStream ()) {
			s.CopyTo (ms);
			ms.Seek (0, SeekOrigin.Begin);

			try {
				return XDocument.Load (ms);
			} catch (XmlException ex) {
				Console.Error.WriteLine ($"Warning: failed to load annotation document '{fileName}' directly from the annotations archive. {ex.Message}");
				Console.Error.WriteLine ("Attempting to fix up and reload");
			}

			try {
				using (var ns = FixAnnotationXML (ms))
					return XDocument.Load (ns);
			} catch (Exception ex) {
				throw new InvalidOperationException ($"Failed to fix up invalid XML in annotation document '{fileName}'. {ex.Message}", ex);
			}
		}
	}
```